### ISSUE TITLE
Multiple Contexts and Transactions

### ISSUE THREAD

**Issue by TOuhrouche** [dotnet/efcore#28880]
I have a domain with two bounded contexts, employees and partners. Each bounded context has its own context. The contexts save operations are coordinated and grouped in a transaction. 

```
        static void Main(string[] args)
        {
            var employeeContext = new EmployeeContext(ConnectionString);
            var partnersContext = new PartnersContext(employeeContext.Database.GetDbConnection());
            var unitOfWork = new UnitOfWork();
            unitOfWork.Update(employeeContext, partnersContext, 1);
            unitOfWork.Update(employeeContext, partnersContext, 2);

        }

        public class UnitOfWork
        {
            public void Update(EmployeeContext employeeContext, PartnersContext partnerContext, int count)
            {
                partnerContext.Partners.Add(new Partner($"John Smith {count}"));
                employeeContext.Persons.Add(new Person() { Name = $"Richard Keno {count}" });

                using var trans = employeeContext.Database.BeginTransaction();
                partnerContext.Database.UseTransaction(trans.GetDbTransaction());
                partnerContext.SaveChanges();
                employeeContext.SaveChanges();
                trans.Commit();
            }
        }
```
A full version of the code is available [here](https://github.com/TOuhrouche/EF-DDD). The first `UnitOfWork.Update` goes through, and the database is updated, but the second call fails with the error below.
![image](https://user-images.githubusercontent.com/1086814/186600941-3e77dba5-ac02-42c1-82e7-2814a3ad448c.png)

You can find more details about the issue [here](https://stackoverflow.com/questions/73477894/ef-core-multiple-contexts-and-transactions/73479511?noredirect=1#comment129761500_73479511)

### Include provider and version information

EF Core version:
Database provider: Microsoft.EntityFrameworkCore.SqlServer  3.1.12
Target framework: netcoreapp3.1
Operating system: Windows 11
IDE: Visual Studio 2022 v 17.0.2

**Comment by TOuhrouche** [comment]
Update: it looks like this is a bug in EF Core 3.1. More about this here: https://stackoverflow.com/a/73483943/782719

**Comment by ajcvickers** [comment]
Fixed in later versions. No plans to fix in 3.1, which is out-of-support before the end of the year.
